src/sql/src/pure.rs

```diff
@@ -281,23 +281,74 @@ pub async fn purify_create_source(
                     _ => bail!("{} is not a postgres connection", item.name()),
                 }
             };
-            let crate::plan::statement::PgConfigOptionExtracted { publication, .. } =
-                options.clone().try_into()?;
+            let crate::plan::statement::PgConfigOptionExtracted {
+                publication,
+                mut text_columns,
+                ..
+            } = options.clone().try_into()?;
             let publication = publication
                 .ok_or_else(|| sql_err!("POSTGRES CONNECTION must specify PUBLICATION"))?;
 
             // verify that we can connect upstream and snapshot publication metadata
             let config = connection
                 .config(&*connection_context.secrets_reader)
                 .await?;
-            let tables = mz_postgres_util::publication_info(&config, &publication).await?;
+
+            let mut text_cols_dict: HashMap<u32, HashSet<String>> = HashMap::new();
```

Comment:
Because of the way we handle PG subsources, we need to know if a column of an unsupported type should actually be allowed because the user specified it in `TEXT COLUMNS`––this is also where we generate the column definitions for `CREATE SUBSOURCE` statements, so we need to know that it should be treated as text. Not really a simple way to avoid doing this.